### PR TITLE
Add `serverComponents.server()` to server-fns plugins when `experimental.islands` is enabled.

### PR DESCRIPTION
**Pull request by jtag05**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Outlined in [1559](https://github.com/solidjs/solid-start/issues/1559)
TL;DR production builds are not adding server action chunks when `experimental.islands` are enabled

## What is the new behavior?
Chunks are generated as expected

### CODE REVIEW

**Comment by jtag05** [comment]
I'm happy to add tests as soon as I can find where they are for the config file. I felt it was best to open this and start the discussion.

**Comment by ryansolid** [comment]
Yeah its ok. This was an oversight in a sense. Like I'm not sure this will work perfectly but lets unblock and see what's broken.
